### ISSUE TITLE
Require a verbose changelog entry for all enhancement bugs

### ISSUE THREAD

**Issue by wilzbach** [dlang-bots/dlang-bot#178]
tl;dr: we should start treating enhancement "bugs" as new features and promote them as such to the users. They won't open the bug link, but will see a full text changelog notice.
Motivation: https://github.com/dlang/dmd/pull/6880#issuecomment-362796817

Looking at the current changelog from https://dlang.org/changelog/pending.html

![image](https://user-images.githubusercontent.com/4370550/35766343-d6374cb4-08d6-11e8-8e29-47cd322a2feb.png)

![image](https://user-images.githubusercontent.com/4370550/35766345-dbbef54c-08d6-11e8-85d7-6ee36de1542b.png)

They all should have received a proper changelog entry. It can be a short one.
Most of them even do, but the following ones were missed:

- Implement contracts without implementation.
- Cannot initialise contextless delegate at compile time
- Parameter type mismatch error message are not very helpful
- datetime toString functions should accept sink
- std.format.formattedRead should work with rvalues

and of course some "enhancements" were actually bugs

**Comment by CyberShadow** [comment]
Another instance: https://github.com/dlang/dmd/pull/12592
